themes/powerline-icon/powerline-icon.base.sh

```diff
@@ -0,0 +1,171 @@
+#! bash oh-my-bash.module
+
+# Define this here so it can be used by all of the Powerline themes
+THEME_CHECK_SUDO=${THEME_CHECK_SUDO:=false}
+
+function set_color {
+  if [[ "${1}" != "-" ]]; then
+    fg="38;5;${1}"
+  fi
+  if [[ "${2}" != "-" ]]; then
+    bg="48;5;${2}"
+    [[ -n "${fg}" ]] && bg=";${bg}"
+  fi
+  echo -e "\[\033[${fg}${bg}m\]"
+}
+
+function __powerline_user_info_prompt {
+  local user_info=""
+  local color=${USER_INFO_THEME_PROMPT_COLOR}
+
+  if [[ "${THEME_CHECK_SUDO}" = true ]]; then
+    # check whether sudo is active for no-password executions
+    if sudo -n cat <<< c3bcc5c 2>&1 | grep -q c3bcc5c; then
+      color=${USER_INFO_THEME_PROMPT_COLOR_SUDO}
+    fi
+  fi
+  case "${POWERLINE_PROMPT_USER_INFO_MODE}" in
+    "sudo")
+      if [[ "${color}" == "${USER_INFO_THEME_PROMPT_COLOR_SUDO}" ]]; then
+        user_info="!"
+      fi
+      ;;
+    *)
+      if [[ -n "${SSH_CLIENT}" ]]; then
+        user_info="${USER_INFO_SSH_CHAR}${USER}@${HOSTNAME}"
+      else
+        user_info="🐧 Linux `date +%X\ %D`"
```

Comment:
This is not a user info. As far as you modify the function content, the code needs to be consistent with the function name. This function returns the user information except for the part that you modified, so the function name should be kept as is. Your modification broke the consistency of the function name and the actual processing. Instead of modifying an unrelated function, you need to prepare a new function if you would like to show the date with a Linux icon.